inflate_p.h

```diff
@@ -211,3 +211,81 @@ static inline void window_output_flush(PREFIX3(stream) *strm) {
     state->whave += out_bytes;
     state->whave = MIN(state->whave, state->wsize);
 }
+
+/* Behave like chunkcopy, but avoid writing beyond of legal output. */
+static inline uint8_t* chunkcopy_safe(uint8_t *out, uint8_t *from, unsigned len, uint8_t *safe) {
+    unsigned safelen = (unsigned)((safe - out) + 1);
+    len = MIN(len, safelen);
+    int olap_src = from >= out && from < out + len;
+    int olap_dst = out >= from && out < from + len;
+
+    /* For all cases without overlap, memcpy is ideal */
+    if (!(olap_src || olap_dst)) {
+        memcpy(out, from, len);
+        return out + len;
+    }
+
+    /* We are emulating a self-modifying copy loop here.  To do this in a way
```

Comment:
Nitpick, but remember that we accept up to 120 char lines in general, and sometimes a little more if there is a clear benefit to readability, such as a 130 char function declaration for example.